applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -60,368 +57,300 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
      */
-    public Flux<ConfigurationSetting> listKeyValues(KeyValueListFilter filter) {
-        return listKeyValues(filter, pageResponseFlux -> pageResponseFlux.map(r -> r.items())
-                                                            .flatMapIterable(i -> i));
-    }
-
-    public <T> Flux<T> listKeyValues(KeyValueListFilter filter, Function<Flux<RestPagedResponse<ConfigurationSetting>>, ? extends Flux<T>> receiver) {
-        Flux<RestPagedResponse<ConfigurationSetting>> p = listSinglePageAsync(filter)
-                       .concatMap(page -> {
-                           String nextPageLink = page.nextLink();
-                           if (nextPageLink == null) {
-                               return Flux.just(page);
-                           }
-                           return Flux.just(page).concatWith(listNextAsync(nextPageLink));
-                       });
-        return receiver.apply(p);
-    }
-
-    public static List<HttpPipelinePolicy> getDefaultPolicies(String connectionString) {
-        final ApplicationConfigCredentials credentials = new ApplicationConfigCredentials(connectionString);
-        // Closest to API goes first, closest to wire goes last.
-        final List<HttpPipelinePolicy> policies = new ArrayList<>();
-
-        policies.add(new UserAgentPolicy(String.format("Azure-SDK-For-Java/%s (%s)", SDK_NAME, SDK_VERSION)));
-        policies.add(new RequestIdPolicy());
-        policies.add(new RetryPolicy());
-        policies.add(new ConfigurationCredentialsPolicy(credentials));
+    public Mono<RestResponse<ConfigurationSetting>> getKeyValue(String key, String label, String etag) {
+        if (key == null || key.isEmpty()) {
+            throw new IllegalArgumentException("Parameter 'key' is required and cannot be null or empty");
+        } else if (label == null) {
+            label = ConfigurationSetting.NULL_LABEL;
+        }
 
-        return policies;
+        return service.getKeyValue(baseUri.toString(), key, label, null, null, null, getETagValue(etag));
     }
 
     /**
-     * Gets all ConfigurationSetting settings.
+     * Deletes the ConfigurationSetting.
      *
-     * @return the Flux&lt;RestPagedResponse&lt;ConfigurationSetting&gt;&gt; object if successful.
+     * @param key The key to delete.
+     * @return the deleted ConfigurationSetting or null if didn't exist.
      */
-    private Flux<RestPagedResponse<ConfigurationSetting>> listSinglePageAsync(KeyValueListFilter filter) {
-        Mono<RestResponse<PageImpl<ConfigurationSetting>>> result;
-        if (filter != null) {
-            result = service.listKeyValues(baseUri.toString(), filter.key(), filter.label(), filter.fields(), filter.acceptDateTime(), filter.range());
-        } else {
-            result = service.listKeyValues(baseUri.toString(), null, null, null, null, null);
+    public Mono<RestResponse<ConfigurationSetting>> deleteKeyValue(String key) {
```

Comment:
Is there value on having this method? Couldn't the users just call `deleteKeyValue(key, null, null)`? 
It will also be consistent with Python and .NET were we only have one implementation for delete

Reply:
I think so, but would be subject to API review. only giving them `delete(key, label, etag)` would force them to understand what a label and etag are. The beauty of C# and python is that they can simply do `delete(key)` and it'll work.

Reply:
Fair point.